src/mlpack/core/data/load_csv.hpp

```diff
@@ -119,33 +205,55 @@ class LoadCSV
     while (std::getline(inFile, line))
     {
       ++rows;
-      // Remove whitespace from either side.
-      boost::trim(line);
-
+      // Remove whitespaces from either side
+      trim(line);
       if (rows == 1)
       {
         // Extract the number of columns.
-        auto findColSize = [&cols](iter_type) { ++cols; };
-        qi::parse(line.begin(), line.end(),
-            stringRule[findColSize] % delimiterRule);
+        std::pair<size_t, size_t> dimen = GetNonNumericMatSize(inFile, delim);
+        cols = dimen.second;
       }
 
       // I guess this is technically a second pass, but that's ok... still the
       // same idea...
       if (MapPolicy::NeedsFirstPass)
       {
         // In this case we must pass everything we parse to the MapPolicy.
-        auto firstPassMap = [&](const iter_type& iter)
+        std::string str(line.begin(), line.end());
+
+        std::stringstream line_stream;
+        std::string token;
+
+        if(line.size() == 0)
+	{
+	  break;
+        }
+
+        line_stream.clear();
+        line_stream.str(line);
+
+        while(line_stream.good())
         {
-          std::string str(iter.begin(), iter.end());
-          boost::trim(str);
+          std::getline(line_stream, token, delim);
+          // Remove whitespace from either side
+	  trim(token);
+
+          if(token[0] == '"' && token[token.size() - 1] != '"')
+          {
+            std::string tok = token;
+
+	    while(token[token.size() - 1] != '"')
+	    {
+	      tok += delim;
+	      std::getline(line_stream, token, delim);
```

Comment:
Okay, so I guess the strategy is if we have an unclosed quote, we keep getting tokens until one of the tokens closes the quote?  I believe this strategy will work; I can't think of any cases where it wouldn't.  (But I didn't think about it *that* hard.)  Let me know if I misunderstood---I'm mostly leaving this comment just to check my own understanding of the code. :)

Reply:
Yeah, this is the strategy we are following here.